spew/config.go

```diff
@@ -98,6 +98,10 @@ type ConfigState struct {
 	// be spewed to strings and sorted by those strings.  This is only
 	// considered if SortKeys is true.
 	SpewKeys bool
+
+	// OmitEmpty specifies that empty struct fields should be omitted.
+	// Empty fields are any that have the zero value for their type.
+	OmitEmpty bool
```

Comment:
I prefer `OmitZeroStructFields` as only struct fields are impacted and I propose to not consider the same way an empty slice/map vs a `nil` one. IMO only `nil` ones should be omitted, just to allow to see the difference between a `nil` (a.k.a non allocated) and an empty item (allocated but empty).
Following the same logic, empty array should not be omitted, so the behavior is consistent with non `nil` empty slice/map as I described above. Moreover, empty arrays are rare, so no need to try to hide them.
`nil` functions can be omitted too.